app/controllers/Checkout.scala

```diff
@@ -96,7 +98,10 @@ object Checkout extends Controller with LazyLogging with ActivityTracking {
 	      product.foreach(prod => trackAnon(SubscriptionRegistrationActivity(MemberData(result, formData, prod))))
       }
 
-      Redirect(routes.Checkout.thankYou()).withSession(session)
+      Ok(Json.obj("redirect" -> routes.Checkout.thankYou().url)).withSession(session);
+    }.recover {
+      case err: soap.Error if err.code == "TRANSACTION_FAILED" => Forbidden
```

Comment:
`Forbidden` looks a bit inaccurate; is it possible to get a more qualified exception? The checkout process goes through a lot of steps, and it's not really evident from the HTTP response that it's the payment step that fails